### ISSUE TITLE
management subnet - private_ip_google_access

### ISSUE THREAD

**Issue by avasseur-pivotal** [pivotal-cf/terraforming-gcp#45]
v0.14.0
The management subnet has the default private ip google access to false

If you uncheck resource config to have the director NOT having public IP adress (a good idea for security ) then it cannot bosh upload stemcell anymore

```
ubuntu@gcp-ops-manager:~/alex$ bosh2 -e pcf upload-stemcell light-bosh-stemcell-3445.17-google-kvm-ubuntu-trusty-go_agent.tgz
Using environment 'https://10.0.2.10' as user 'director' (bosh.*.read, openid, bosh.*.admin, bosh.read, bosh.admin)

######################################################### 100.00% 166.27 KB/s 0s
Task 23

Task 23 | 22:20:39 | Update stemcell: Extracting stemcell archive (00:00:00)
Task 23 | 22:20:39 | Update stemcell: Verifying stemcell manifest (00:00:00)
Task 23 | 22:20:39 | Update stemcell: Checking if this stemcell already exists (00:00:00)
Task 23 | 22:20:39 | Update stemcell: Uploading stemcell bosh-google-kvm-ubuntu-trusty-go_agent/3445.17 to the cloud (00:02:17)
                   L Error: CPI error 'Bosh::Clouds::CloudError' with message 'Creating stemcell: Creating Google Image from URL: Failed to create Google Image: Post https://www.googleapis.com/compute/v1/projects/fe-avasseur/global/images?alt=json: dial tcp 64.233.166.95:443: i/o timeout' in 'create_stemcell' CPI method
Task 23 | 22:22:56 | Error: CPI error 'Bosh::Clouds::CloudError' with message 'Creating stemcell: Creating Google Image from URL: Failed to create Google Image: Post https://www.googleapis.com/compute/v1/projects/fe-avasseur/global/images?alt=json: dial tcp 64.233.166.95:443: i/o timeout' in 'create_stemcell' CPI method

Task 23 Started  Sat Dec  9 22:20:39 UTC 2017
Task 23 Finished Sat Dec  9 22:22:56 UTC 2017
Task 23 Duration 00:02:17
Task 23 error
```

Change in networks.tf is required - imho with a better default to true
```
  ~ google_compute_subnetwork.management-subnet
      private_ip_google_access:        "false" => "true"
```

screenshot
<img width="523" alt="screen shot 2017-12-09 at 23 35 03" src="https://user-images.githubusercontent.com/5575656/33800111-4dd5bc38-dd3a-11e7-8a51-3e8c2881b93c.png">

**Comment by jasonbisson** [comment]
I've ran into this issue too and interested if this will be considered.

**Comment by nthomson-pivotal** [comment]
Based on what I've seen in the last couple of days there are related issues with respect to PKS. The k8s worker nodes do not have public IPs, which means:

- When creating a cluster the BOSH tasks would fail when building the worker VMs, the issue was the same as mentioned above (Google API access). I was able to resolve this by manually activating private access on the appropriate GCP sub-network.
- After building a k8s clusters, when deploying images from DockerHub I encountered more failures which I solved by adding a NAT gateway.

My current thoughts are that we really need a NAT gateway, at least optionally.
